mlflow/spark/__init__.py

```diff
@@ -992,26 +992,66 @@ def predict(
             # Spark model uses "prediction" as default model inference output column name.
             return self.spark_model.transform(pandas_df)["prediction"]
 
+        # Convert List[np.float64] / np.array[np.float64] type to List[float] type,
+        # otherwise it will break `spark.createDataFrame` column type inferring.
+        if self.signature.inputs:
+            for col_spec in self.signature.inputs.inputs:
+                if isinstance(col_spec.type, Array) and col_spec.type.is_sparkml_vector:
```

Comment:
we'll need a guard here for the additional field for users who load an older model in an updated version of MLflow. Let's verify that `col_spec.type` has the attribute `is_sparkml_vector` before doing a direct access of it.

Reply:
oh yes good catch. we should keep compatibility.